File: angular/src/app/models/claim.ts
import { User } from "./user";
import { Userstory } from "./userstory";

export interface Claim {
    id: number;
    title: string;
    content: number;
    image: string;
    employeeCode: string;

    userstory: Userstory;
    user: User;
  }

File: spring/src/main/java/tn/esprit/codemasters/entity/Claim.java
package tn.esprit.codemasters.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import tn.esprit.codemasters.entity.user.User;

@Entity
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level=AccessLevel.PRIVATE)
public class Claim {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id;
    String title;
    String content;

    //h
    @ManyToOne
    UserStory userstory;

    @ManyToOne
    User user;
}
